src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -2076,6 +2076,13 @@ namespace winrt::TerminalApp::implementation
         const auto windowId{ args.Window() };
         if (!windowId.empty())
         {
+            // if the windowId is the same as our name, do nothing
+            if (windowId == WindowProperties().WindowName() ||
+                windowId == winrt::to_hstring(WindowProperties().WindowId()))
```

Comment:
Hmmmmm okay, but it does feel a bit weird

Reply:
Can you explain why it needs to compare against both properties? I don't quite get that...

Reply:
@lhecker so, the deal is that we generally treat `"window"` as a stealth union. Users can either specify a window name or a window ID in that property. So actions like

```json
        { "keys": "f1", "command": { "action": "moveTab", "window": "1" }, },
        { "keys": "f2", "command": { "action": "moveTab", "window": "foo" }, },
```

both work. 

We don't know whether the string was the ID, or the name, so we have to check both.

Reply:
:cry: why can't we just be like, `if(this == otherwindae)`